File: api/src/test/java/net/blueshell/api/factory/dto/GuestDTOFactory.java
package net.blueshell.api.factory.dto;

import com.github.javafaker.Faker;
import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.GuestDTO;
import org.springframework.stereotype.Component;

import java.util.Locale;

/**
 * Factory for GuestDTO test instances.
 */
@Component
@RequiredArgsConstructor
public class GuestDTOFactory extends BaseDtoFactory<GuestDTO> {

    @Override
    public Class<GuestDTO> targetType() {
        return GuestDTO.class;
    }

    @Override
    public GuestDTO createBasic() {
        GuestDTO dto = new GuestDTO();
        dto.setCreatedAt(now());
        dto.setName("Guest " + nextId());
        dto.setDiscord("guest");
        dto.setEmail(email("guest"));
        dto.setPhoneNumber("+31651319571");
        dto.setAccessToken(unique("acct"));
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/GuestDTOFactory.kt
package net.blueshell.api.factory.dto

import net.blueshell.api.feature.event.dto.GuestDTO
import org.springframework.stereotype.Component

/**
 * Factory for GuestDTO test instances.
 */
@Component
class GuestDTOFactory : BaseDtoFactory<GuestDTO>() {

    override fun targetType(): Class<GuestDTO> = GuestDTO::class.java

    override fun createBasic(): GuestDTO {
        val dto = GuestDTO()
        dto.createdAt = now()
        dto.name = "Guest ${nextId()}"
        dto.discord = "guest"
        dto.email = email("guest")
        dto.phoneNumber = "+31651319571"
        return dto
    }
}
